osu.Game/Screens/OnlinePlay/Multiplayer/Spectate/MultiSpectatorScreen.cs

```diff
@@ -215,6 +219,11 @@ protected override void StartGameplay(int userId, GameplayState gameplayState)
         protected override void EndGameplay(int userId)
         {
             RemoveUser(userId);
+
+            var instance = instances.Single(i => i.UserId == userId);
```

Comment:
Are you 100% sure this will always find a result? `RemoveUser` above gracefully handles the case where `userId` is not tracked, so it's best to make sure. Or use `SingleOrDefault` for safety.